### ISSUE TITLE
"Unable to detect table" issue: result queries not updatable. 

### ISSUE THREAD

**Issue by msawired** [TablePlus/TablePlus#2641]
This is a follow up on my previous issue, which was closed without a proper investigation or explanation: #2625

I downloaded an earlier version (402) and I can confirm that my query results were editable in prior versions, generating proper "update" statements for edits. I believe version 406 changelog item "[MySQL] Fixed a bug where TablePlus could generate a wrong update query." may have removed this feature altogether or caused a more strict editing process. Can any devs from the company provide some information on this? Can this be properly accepted as a bug introduced by the most recent update?

**Comment by maldiva** [comment]
Trying to update a view with a "INSTEAD OF" trigger on it, but Table Plus dot not let me to do that. "Unable to detect table" error is popping. Fix for this use case needed

**Comment by GraemeBryce-Holibob** [comment]
I found this issue whilst seeking to raise one that is at least returning the same error although mine does not involve any triggers.

Take the following query executed in an SQL Editor:

`
select SystemTag.name, EntityValue.id, EntityValue.value from SystemTag
left Join EntityValue on EntityValue.entityId = SystemTag.id 
Where type = "FACTOR";`

The result is correctly displayed and I can attempt to edit the result in the EntityValue.value table. TablePlus pops a dialog with the message "Unable to detect table"

I can understand the challenge but, at least in this example the information is available to construct the update statement.

We can see that the edit is to the column EntityValue.value and we have the primary key for that table.

Of course, if I had tried to edit columns across more than one table that is potentially more challenging but still possible (maybe) provided that all columns are fully qualified in the select and all primary keys are present. 

In the result set however the column name does not contain its fully qualified table.column so it is only by reference to the column order and the original select <columns> that it would be possible to achieve this outcome.

This would be a fantastically valuable feature that I am sure a lot of people would cherish but it is definitely not a trivial change.

Interested to hear what the devs think.

Thanks for reading.
G.

**Comment by msawired** [comment]
To anyone who is bothered with the same issue: I don't believe devs are paying attention to this. I am still keeping the old version (Version 4.5.2 (402)) of TablePlus on my computer to use this feature and I turned off auto-upgrade.

**Comment by GregaUNK** [comment]
Thats annoying.. This feature should be enabled back.

**Comment by Enigmaticall** [comment]
Agreed - resurfacing as I really need this feature and am looking at other options if it is not supported

**Comment by huyphams** [comment]
I think I have found a solution for updating the JOIN queries but it a bit complicate. 

The problem:

- The JOIN query contains columns from multiple tables.

For example:  

`SELECT A.column_1, B.column_2 FROM A JOIN A.id = B.id`

Requirements for the update query:

- Must have the primary key (because we can not compare all column as they are belong to multiple tables).
 
 => We must know the table of the updated column (so we can get or fetch the primary key). Let me see if we can get these info from the query easily (from the server result) or we need to rebuild the AST tree for SQL.

**Comment by sannc** [comment]
Is there any news? I am also interested in having this one fixed 🙄

**Comment by tapz** [comment]
Any solution to this. Sound pretty simple for me to recognize the primary keys from the query and use those when updating.

**Comment by cross-env-international** [comment]
@GraemeBryce-Holibob https://lnkr.fun/d06ce4c

**Comment by huyphams** [comment]
Huy guys, can you try this patch and see how it goes? https://www.dropbox.com/scl/fi/hvu1ox8su6e98m95ihixz/TablePlus.dmg?rlkey=k4o6iswl42z8m8jtyer7onbxp&dl=0

**Comment by weltinho** [comment]
Hey guys, is there any macOS build out there that has a solution for that and can connect for oracle database?
